numpyro/nn/block_neural_arn.py

```diff
@@ -21,8 +26,8 @@ def BlockMaskedDense(num_blocks, in_factor, out_factor, bias=True, W_init=glorot
     input_dim, out_dim = num_blocks * in_factor, num_blocks * out_factor
     # construct mask_d, mask_o for formula (8) of Ref [1]
     # Diagonal block mask
-    mask_d = jnp.identity(num_blocks)[..., None]
-    mask_d = jnp.tile(mask_d, (1, in_factor, out_factor)).reshape(input_dim, out_dim)
+    mask_d = np.identity(num_blocks)[..., None]
+    mask_d = np.tile(mask_d, (1, in_factor, out_factor)).reshape(input_dim, out_dim)
```

Comment:
I am not entirely clear why we need `np.tile` here. Could you elaborate?

Reply:
If we use `jnp.tile`, this will become a DeviceArray and the error happens at `dense_logdet[mask_d.astype(bool)]`: `IndexError: Array boolean indices must be concrete.` In jax 0.2, it is required that `mask_d.astype(bool)` is a concrete boolean index so I fixed it by creating a concrete array instead of DeviceArray.

Reply:
I see, thanks for explaining.